src/cleed++/Search/SearchAtom.hh

```
@@ -0,0 +1,81 @@
+#ifdef __cplusplus /* if this is a C++ compiler then use C linkage */
+
+#include <vector>
+#include <cstddef>
+#include <string>
+
+#include <Core/Atom.hh>
+#include "search_def.h"
+
+namespace cleed {
+
+  class SearchAtom : public search_atom {
```

Clang (14.0.6)

[warning] destructor of 'SearchAtom' is public and non-virtual

(cppcoreguidelines-virtual-class-destructor)